File: python-algorithm/bubble_sort.py
## filename: bubble_sort.py
import random
rng = 20 # default value
rng = int(input("Please input a value of range: "))
lst = [random.randrange(0, rng) for idx in range(rng)]

# bubble sort
times = rng
copy_lst = lst[:]
for i in range(times):
    for seq in range(rng - 1):
        if lst[seq] > lst[seq + 1]:
            tmp = lst[seq]
            lst[seq] = lst[seq + 1]
            lst[seq + 1] = tmp
    rng-=1

# print the original-list and sorted-list
print("=" * 40)
print("The original-list:\n", copy_lst)
print("The sorted-list:\n", lst)
copy_lst.sort()
print("Compare with result sorted by list.sort():\n", copy_lst)
print("Is the sort-algorithm right?\n", copy_lst == lst)
print("--> THIS IS BUBBLE SORT! <--")

File: clang-algorithm/sort_algorithm/bubble_sort/bubble_sort.c
// bubble sort
#include <stdio.h>
int main(void){
    int data[5] = {80, 77, 54, 19, 11};
    int i, j, t;
    int len = 5;

    // 排序前数据输出
    for (i=0; i<len; i++){
        printf("%d ", data[i]);
    }
    putchar('\n');

    // len个数据只需要len-1趟排序
    for (i = 0; i < len - 1; i++){
        // 每趟比较次数不断减少
        for (j = 0; j < len - i - 1; j++){
            if (data[j] > data[j+1]){
                // 交换两个变量的值
                t = data[j];
                data[j] = data[j+1];
                data[j+1] = t;
            }
       }
    }

    // 排序后输出
    for (i=0; i<len; i++){
        printf("%d ", data[i]);
    }
    putchar('\n');

    return 0;
} 
